### PR TITLE
media-gfx/zbar: add 0.23.93 with python3_12 support

### PR DESCRIPTION
**Pull request by alanswanson**
Removed patches from ebuild with equivalent upstream commits and tested with python 3.12.

Closes: https://bugs.gentoo.org/929630

---

Please check all the boxes that apply:

- [X] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [X] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [X] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [X] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: media-gfx/zbar/zbar-0.23.93.ebuild [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+				myeconfargs+=( --with-java-unit )
+				CLASSPATH+=":$(java-pkg_getjar --build-only junit-4 junit.jar)"
+				CLASSPATH+=":$(java-pkg_getjar --build-only hamcrest-core-1.3 hamcrest-core.jar)"
+			fi
+		fi
+
+		if use qt5; then
+			myeconfargs+=(
+				--with-qt
+				--with-qt5
```
Comment (puleglot) [inline media-gfx/zbar/zbar-0.23.93.ebuild:178]: AFAIS there is no such configure option in 0.23.93. zbar defaults to QT5 and there is experimental support for QT6 (`--with-qt6`)

Comment (alanswanson) [reply media-gfx/zbar/zbar-0.23.93.ebuild]: Removed "qt5" option leaving "qt" only. Note there is no actual qt6 support in zbar, just a "prepare to support" commit to configure.ac which seems pointless before qt6 support is actually added to the code...

File: media-gfx/zbar/zbar-0.23.93.ebuild [hunk 2]
```diff
@@ -0,0 +1,230 @@
+# Copyright 1999-2023 Gentoo Authors
```
Comment (juippis) [inline media-gfx/zbar/zbar-0.23.93.ebuild:1]:
```suggestion
# Copyright 1999-2024 Gentoo Authors
```
Use tools like **pkgdev** to commit to fix these automatically.

**Review by juippis — COMMENTED** [review]
There's something weird happening with this package:
```
>>> Source configured.
>>> Compiling source in /var/tmp/portage/media-gfx/zbar-0.23.93/work/zbar-0.23.93 ...
 * abi_x86_64.amd64: running multilib-minimal_abi_src_compile
make -j16 
/usr/bin/mkdir -p qt 
Iqt -DQT_WIDGETS_LIB -I/usr/include/qt5/QtWidgets -I/usr/include/qt5 -DQT_X11EXTRAS_LIB -I/usr/include/qt5/QtX11Extras -DQT_GUI_LIB -I/usr/include/qt5/QtGui -DQT_CORE_LIB -I/usr/include/qt5/QtCore -I/var/tmp/portage/media-gfx/zbar-0.23.93/work/zbar-0.23.93/include /var/tmp/portage/media-gfx/zbar-0.23.93/work/zbar-0.23.93/qt/QZBarThread.h -o qt/moc_QZBarThread.cpp
/bin/bash: line 1: Iqt: command not found
make: [Makefile:2486: qt/moc_QZBarThread.cpp] Error 127 (ignored)
Iqt -DQT_WIDGETS_LIB -I/usr/include/qt5/QtWidgets -I/usr/include/qt5 -DQT_X11EXTRAS_LIB -I/usr/include/qt5/QtX11Extras -DQT_GUI_LIB -I/usr/include/qt5/QtGui -DQT_CORE_LIB -I/usr/include/qt5/QtCore -I/var/tmp/portage/media-gfx/zbar-0.23.93/work/zbar-0.23.93/include /var/tmp/portage/media-gfx/zbar-0.23.93/work/zbar-0.23.93/include/zbar/QZBar.h -o qt/moc_QZBar.cpp
/bin/bash: line 1: Iqt: command not found
make: [Makefile:2490: qt/moc_QZBar.cpp] Error 127 (ignored)
...
...
cc1plus: fatal error: /var/tmp/portage/media-gfx/zbar-0.23.93/work/zbar-0.23.93/qt/moc_QZBar.cpp: No such file or directory
compilation terminated.
make[2]: *** [Makefile:1674: qt/libzbarqt_la-moc_QZBar.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
cc1plus: fatal error: /var/tmp/portage/media-gfx/zbar-0.23.93/work/zbar-0.23.93/qt/moc_QZBarThread.cpp: No such file or directory
compilation terminated.
make[2]: *** [Makefile:1681: qt/libzbarqt_la-moc_QZBarThread.lo] Error 1
```
```
Making all in gtk 
make[2]: Entering directory '/var/tmp/portage/media-gfx/zbar-0.23.93/work/zbar-0.23.93-abi_x86_64.amd64/gtk'
g-fatal-warnings --prefix=zbar_marshal \
    --body /var/tmp/portage/media-gfx/zbar-0.23.93/work/zbar-0.23.93/gtk/zbarmarshal.list > zbarmarshal.c
g-fatal-warnings --prefix=zbar_marshal \
    --header /var/tmp/portage/media-gfx/zbar-0.23.93/work/zbar-0.23.93/gtk/zbarmarshal.list > zbarmarshal.h
/bin/bash: line 1: g-fatal-warnings: command not found
/bin/bash: line 1: g-fatal-warnings: command not found
make[2]: [Makefile:912: zbarmarshal.c] Error 127 (ignored)
make[2]: [Makefile:908: zbarmarshal.h] Error 127 (ignored)
```

**Comment by puleglot** [comment]
Two upstream patches are needed to fix qt:
https://github.com/mchehab/zbar/commit/368571ffa1a0f6cc41f708dd0d27f9b6e9409df8
https://github.com/mchehab/zbar/commit/a549566ea11eb03622bd4458a1728ffe3f589163

**Comment by puleglot** [comment]
> ```
> Making all in gtk 
> make[2]: Entering directory '/var/tmp/portage/media-gfx/zbar-0.23.93/work/zbar-0.23.93-abi_x86_64.amd64/gtk'
> g-fatal-warnings --prefix=zbar_marshal \
>     --body /var/tmp/portage/media-gfx/zbar-0.23.93/work/zbar-0.23.93/gtk/zbarmarshal.list > zbarmarshal.c
> g-fatal-warnings --prefix=zbar_marshal \
>     --header /var/tmp/portage/media-gfx/zbar-0.23.93/work/zbar-0.23.93/gtk/zbarmarshal.list > zbarmarshal.h
> /bin/bash: line 1: g-fatal-warnings: command not found
> /bin/bash: line 1: g-fatal-warnings: command not found
> make[2]: [Makefile:912: zbarmarshal.c] Error 127 (ignored)
> make[2]: [Makefile:908: zbarmarshal.h] Error 127 (ignored)
> ```

Hmm.. it does work with dash
```
Making all in gtk
make[2]: Entering directory '/var/tmp/portage/media-gfx/zbar-0.23.93/work/zbar-0.23.93-abi_x86_64.amd64/gtk'
/usr/bin/glib-genmarshal --g-fatal-warnings --prefix=zbar_marshal \
    --body /var/tmp/portage/media-gfx/zbar-0.23.93/work/zbar-0.23.93/gtk/zbarmarshal.list > zbarmarshal.c
/usr/bin/glib-genmarshal --g-fatal-warnings --prefix=zbar_marshal \
    --header /var/tmp/portage/media-gfx/zbar-0.23.93/work/zbar-0.23.93/gtk/zbarmarshal.list > zbarmarshal.h
```

**Comment by puleglot** [comment]
And `CONFIG_SHELL=/bin/bash` doesn't make any difference. Still works

**Comment by juippis** [comment]
Could you include those patches in this version bump and I'll try again?

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**